src/Symfony/Component/Process/Process.php

```diff
@@ -1653,9 +1653,15 @@ function ($m) use (&$envBackup, &$varCache, &$varCount, $uid) {
 
                 $value = str_replace(array('!LF!', '"^!"', '"^%"', '"^^"', '""'), array("\n", '!', '%', '^', '"'), $value);
                 $value = preg_replace('/(\\\\*)"/', '$1$1\\"', $value);
-
+                $value = "\"$value\"";
                 $var = $uid.++$varCount;
-                putenv("$var=\"$value\"");
+
+                if ($env === null) {
+                    putenv("$var=$value");
+                } else {
+                    $env[$var] = $value;
```

Comment:
@nicolas-grekas 
I thought the same at the beginning but the argument needs to be escaped even it's passed using the `!var!` notation (and stored inside the env vars). The tests don't pass without the quotes. BTW I am not exactly sure why it's done using the `!var!` workarround. Looks like a clever trick and perhaps you are it's author (due to github history/annotations). Is it somehow related to how inconsistently Windows commands handle unescaping of double quotes? (Should be mostly triple quotes and i've found an article describing it but it's not exactly specified so some commands handle it differently). I'm just curious :-)